sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2349,7 +2350,31 @@ class Analyzer(
   }
 
   /**
-   * Replace [[TimeZoneAwareExpression]] without timezone id by its copy with session local
+   * Replace unresolved expressions in grouping keys with resolved ones in SELECT clauses.
+   */
+  object ResolveAggAliasInGroupBy extends Rule[LogicalPlan] {
+
+    override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperators {
+      case agg @ Aggregate(groups, aggs, child)
+          if conf.groupByAliases && child.resolved && groups.exists(!_.resolved) =>
+        agg.copy(groupingExpressions = groups.map {
+          case u: UnresolvedAttribute =>
+            val resolvedAgg = aggs.find(ne => resolver(ne.name, u.name))
+            // Check if no aggregate function exists in GROUP BY
```

Comment:
Without this entry, the analysis is passed and an exception is thrown in runtime;
```
scala> val df = Seq((1, 0), (2, 0)).toDF("k", "v").createOrReplaceTempView("t")
scala> sql("""SELECT count("v") AS key FROM t GROUP BY key""").show

Caused by: java.lang.UnsupportedOperationException: Cannot evaluate expression: count(1)
  at org.apache.spark.sql.catalyst.expressions.Unevaluable$class.doGenCode(Expression.scala:226)
  at org.apache.spark.sql.catalyst.expressions.aggregate.AggregateExpression.doGenCode(interfaces.scala:87)
  at org.apache.spark.sql.catalyst.expressions.Expression$$anonfun$genCode$2.apply(Expression.scala:106)
  at org.apache.spark.sql.catalyst.expressions.Expression$$anonfun$genCode$2.apply(Expression.scala:103)
  at scala.Option.getOrElse(Option.scala:121)
```
Move this check in `CheckAnalysis`?